#VRML_SIM V8.0.1 utf8
WorldInfo {
  info [
    "Simulation of the YAMOR modular robot (BIRG)."
    "The YAMOR robot goes through a sequence of module deconnections and reconnections while demonstrating two different types of locomotion."
    "Author: Yvan Bourquin"
    "Biologically Inspired Robotics Group (BIRG)"
    "Swiss Federal Institute of Technology (EPFL), Lausanne"
    "Date: 29 June 2006"
    ""
    "Publication:"
    "R. Moeckel, C. Jaquier, K. Drapel, E. Dittrich, A. Upegui, and A.J. Ijspeert."
    "Exploring adaptive locomotion with YaMoR, a novel autonomous modular robot with Bluetooth interface."
    "Industrial Robot, 33(4):285-290, 2006."
  ]
  title "Modular Robot"
  basicTimeStep 8
  contactProperties [
    ContactProperties {
      coulombFriction [
        0.5
      ]
    }
  ]
}
Viewpoint {
  orientation 0.0950926 0.971964 0.215041 3.95723
  position -0.987991 0.674055 -0.704196
  follow "module_4"
}
Background {
  skyColor [
    0.4 0.7 1
  ]
}
DirectionalLight {
  ambientIntensity 1
  direction 1.2 -0.7 -0.5
  castShadows TRUE
}
Floor {
  size 100 100
}
DEF MODULE_1 Robot {
  translation 9.51732e-08 0.0269856 -2.05308e-07
  rotation -1.37759e-05 1.38288e-05 1 1.5708
  children [
    DEF BODY_GROUP Group {
      children [
        DEF BOX_SHAPE Shape {
          appearance DEF PLASTIC Appearance {
            material Material {
              diffuseColor 0 1 0
            }
          }
          geometry Box {
            size 0.054 0.0482 0.06
          }
        }
        DEF CYLINDER_TRANS Transform {
          translation 0 0 -0.03
          children [
            Shape {
              appearance USE PLASTIC
              geometry Cylinder {
                height 0.0482
                radius 0.027
              }
            }
          ]
        }
      ]
    }
    DEF REAR_CONNECTOR Connector {
      translation 0 0 0.03
      children [
        DEF ROUND_SHAPE Transform {
          rotation 1 0 0 1.5708
          children [
            Shape {
              appearance Appearance {
                material Material {
                  diffuseColor 1 1 1
                }
              }
              geometry Cylinder {
                height 0.0005
                radius 0.0075
              }
            }
          ]
        }
      ]
      name "rear_connector"
      isLocked TRUE
      distanceTolerance 0.05
      axisTolerance 0.5
      rotationTolerance 0.5
    }
    DEF HINGE_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        position 2.11893e-09
        axis 0 1 0
        anchor 0 0 -0.03
      }
      device [
        RotationalMotor {
          name "motor"
          maxVelocity 3.9667
          minPosition -1.5708
          maxPosition 1.5708
          maxTorque 1.1
        }
      ]
      endPoint Solid {
        translation -6.2784e-08 3.8508e-18 -0.03
        rotation 3.43781e-10 1 1.30763e-07 2.11893e-09
        children [
          DEF GREY_BOX_TRANS Transform {
            translation 0 0 -0.0135
            children [
              Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 0.752941 0.752941 0.752941
                  }
                }
                geometry Box {
                  size 0.015 0.0612 0.047
                }
              }
            ]
          }
          DEF FRONT_CONNECTOR Connector {
            translation 0 0 -0.037
            rotation 0 1 0 3.1416
            children [
              USE ROUND_SHAPE
            ]
            name "front_connector"
            isLocked TRUE
            distanceTolerance 0.05
            axisTolerance 0.5
            rotationTolerance 0.5
          }
        ]
        boundingObject USE GREY_BOX_TRANS
        physics Physics {
          density -1
          mass 0.016
        }
      }
    }
  ]
  name "module_1"
  boundingObject USE BODY_GROUP
  physics DEF BODY_PHYSICS Physics {
    density -1
    mass 0.17
  }
  controller "LocomotionController"
}
DEF MODULE_2 Robot {
  translation 9.90893e-08 0.0269875 0.0969998
  rotation -1.01248e-05 1.01627e-05 1 1.5708
  children [
    USE BODY_GROUP
    DEF REAR_CONNECTOR Connector {
      translation 0 0 0.03
      children [
        DEF ROUND_SHAPE Transform {
          rotation 1 0 0 1.5708
          children [
            Shape {
              appearance Appearance {
                material Material {
                  diffuseColor 1 1 1
                }
              }
              geometry Cylinder {
                height 0.0005
                radius 0.0075
              }
            }
          ]
        }
      ]
      name "rear_connector"
      isLocked TRUE
      distanceTolerance 0.05
      axisTolerance 0.5
      rotationTolerance 0.5
    }
    DEF HINGE_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        position -2.17511e-08
        axis 0 1 0
        anchor 0 0 -0.03
      }
      device [
        RotationalMotor {
          name "motor"
          maxVelocity 3.9667
          minPosition -1.5708
          maxPosition 1.5708
          maxTorque 1.1
        }
      ]
      endPoint Solid {
        translation 8.88768e-08 -1.06117e-10 -0.03
        rotation 0.00037819 1 -0.00013173 -2.17511e-08
        children [
          DEF GREY_BOX_TRANS Transform {
            translation 0 0 -0.0135
            children [
              Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 0.752941 0.752941 0.752941
                  }
                }
                geometry Box {
                  size 0.015 0.0612 0.047
                }
              }
            ]
          }
          DEF FRONT_CONNECTOR Connector {
            translation 0 0 -0.037
            rotation 0 1 0 3.1416
            children [
              USE ROUND_SHAPE
            ]
            name "front_connector"
            isLocked TRUE
            distanceTolerance 0.05
            axisTolerance 0.5
            rotationTolerance 0.5
          }
        ]
        boundingObject USE GREY_BOX_TRANS
        physics Physics {
          density -1
          mass 0.016
        }
      }
    }
  ]
  name "module_2"
  boundingObject USE BODY_GROUP
  physics USE BODY_PHYSICS
  controller "LocomotionController"
}
DEF MODULE_3 Robot {
  translation 1.01492e-07 0.0269887 0.194
  rotation -6.49881e-06 6.52192e-06 1 1.5708
  children [
    USE BODY_GROUP
    DEF REAR_CONNECTOR Connector {
      translation 0 0 0.03
      children [
        DEF ROUND_SHAPE Transform {
          rotation 1 0 0 1.5708
          children [
            Shape {
              appearance Appearance {
                material Material {
                  diffuseColor 1 1 1
                }
              }
              geometry Cylinder {
                height 0.0005
                radius 0.0075
              }
            }
          ]
        }
      ]
      name "rear_connector"
      isLocked TRUE
      distanceTolerance 0.05
      axisTolerance 0.5
      rotationTolerance 0.5
    }
    DEF HINGE_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        position -5.80179e-08
        axis 0 1 0
        anchor 0 0 -0.03
      }
      device [
        RotationalMotor {
          name "motor"
          maxVelocity 3.9667
          minPosition -1.5708
          maxPosition 1.5708
          maxTorque 1.1
        }
      ]
      endPoint Solid {
        translation 1.19394e-07 -1.41515e-10 -0.03
        rotation 0.000367017 1 -6.5848e-05 -5.80179e-08
        children [
          DEF GREY_BOX_TRANS Transform {
            translation 0 0 -0.0135
            children [
              Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 0.752941 0.752941 0.752941
                  }
                }
                geometry Box {
                  size 0.015 0.0612 0.047
                }
              }
            ]
          }
          DEF FRONT_CONNECTOR Connector {
            translation 0 0 -0.037
            rotation 0 1 0 3.1416
            children [
              USE ROUND_SHAPE
            ]
            name "front_connector"
            isLocked TRUE
            distanceTolerance 0.05
            axisTolerance 0.5
            rotationTolerance 0.5
          }
        ]
        boundingObject USE GREY_BOX_TRANS
        physics Physics {
          density -1
          mass 0.016
        }
      }
    }
  ]
  name "module_3"
  boundingObject USE BODY_GROUP
  physics USE BODY_PHYSICS
  controller "LocomotionController"
}
DEF MODULE_4 Robot {
  translation 1.02546e-07 0.0269893 0.291
  rotation -2.89283e-06 2.90109e-06 1 1.5708
  children [
    USE BODY_GROUP
    DEF REAR_CONNECTOR Connector {
      translation 0 0 0.03
      children [
        DEF ROUND_SHAPE Transform {
          rotation 1 0 0 1.5708
          children [
            Shape {
              appearance Appearance {
                material Material {
                  diffuseColor 1 1 1
                }
              }
              geometry Cylinder {
                height 0.0005
                radius 0.0075
              }
            }
          ]
        }
      ]
      name "rear_connector"
      isLocked TRUE
      distanceTolerance 0.05
      axisTolerance 0.5
      rotationTolerance 0.5
    }
    DEF HINGE_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        position -8.616e-08
        axis 0 1 0
        anchor 0 0 -0.03
      }
      device [
        RotationalMotor {
          name "motor"
          maxVelocity 3.9667
          minPosition -1.5708
          maxPosition 1.5708
          maxTorque 1.1
        }
      ]
      endPoint Solid {
        translation 7.49437e-08 -9.5103e-11 -0.03
        rotation 0.00036535 1 -2.9801e-05 -8.616e-08
        children [
          DEF GREY_BOX_TRANS Transform {
            translation 0 0 -0.0135
            children [
              Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 0.752941 0.752941 0.752941
                  }
                }
                geometry Box {
                  size 0.015 0.0612 0.047
                }
              }
            ]
          }
          DEF FRONT_CONNECTOR Connector {
            translation 0 0 -0.037
            rotation 0 1 0 3.1416
            children [
              USE ROUND_SHAPE
            ]
            name "front_connector"
            isLocked TRUE
            distanceTolerance 0.05
            axisTolerance 0.5
            rotationTolerance 0.5
          }
        ]
        boundingObject USE GREY_BOX_TRANS
        physics Physics {
          density -1
          mass 0.016
        }
      }
    }
  ]
  name "module_4"
  boundingObject USE BODY_GROUP
  physics USE BODY_PHYSICS
  controller "LocomotionController"
}
DEF MODULE_5 Robot {
  translation 1.02352e-07 0.0269893 0.388
  rotation 7.06303e-07 -7.12877e-07 1 1.5708
  children [
    USE BODY_GROUP
    DEF REAR_CONNECTOR Connector {
      translation 0 0 0.03
      children [
        DEF ROUND_SHAPE Transform {
          rotation 1 0 0 1.5708
          children [
            Shape {
              appearance Appearance {
                material Material {
                  diffuseColor 1 1 1
                }
              }
              geometry Cylinder {
                height 0.0005
                radius 0.0075
              }
            }
          ]
        }
      ]
      name "rear_connector"
      isLocked TRUE
      distanceTolerance 0.05
      axisTolerance 0.5
      rotationTolerance 0.5
    }
    DEF HINGE_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        position -9.51477e-08
        axis 0 1 0
        anchor 0 0 -0.03
      }
      device [
        RotationalMotor {
          name "motor"
          maxVelocity 3.9667
          minPosition -1.5708
          maxPosition 1.5708
          maxTorque 1.1
        }
      ]
      endPoint Solid {
        translation -7.57278e-09 1.07155e-13 -0.03
        rotation 0.000350823 1 2.75414e-08 -9.51477e-08
        children [
          DEF GREY_BOX_TRANS Transform {
            translation 0 0 -0.0135
            children [
              Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 0.752941 0.752941 0.752941
                  }
                }
                geometry Box {
                  size 0.015 0.0612 0.047
                }
              }
            ]
          }
          DEF FRONT_CONNECTOR Connector {
            translation 0 0 -0.037
            rotation 0 1 0 3.1416
            children [
              USE ROUND_SHAPE
            ]
            name "front_connector"
            isLocked TRUE
            distanceTolerance 0.05
            axisTolerance 0.5
            rotationTolerance 0.5
          }
        ]
        boundingObject USE GREY_BOX_TRANS
        physics Physics {
          density -1
          mass 0.016
        }
      }
    }
  ]
  name "module_5"
  boundingObject USE BODY_GROUP
  physics USE BODY_PHYSICS
  controller "LocomotionController"
}
DEF MODULE_6 Robot {
  translation 1.00908e-07 0.0269888 0.485
  rotation 4.31407e-06 -4.33549e-06 1 1.5708
  children [
    USE BODY_GROUP
    DEF REAR_CONNECTOR Connector {
      translation 0 0 0.03
      children [
        DEF ROUND_SHAPE Transform {
          rotation 1 0 0 1.5708
          children [
            Shape {
              appearance Appearance {
                material Material {
                  diffuseColor 1 1 1
                }
              }
              geometry Cylinder {
                height 0.0005
                radius 0.0075
              }
            }
          ]
        }
      ]
      name "rear_connector"
      isLocked TRUE
      distanceTolerance 0.05
      axisTolerance 0.5
      rotationTolerance 0.5
    }
    DEF HINGE_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        position -8.19638e-08
        axis 0 1 0
        anchor 0 0 -0.03
      }
      device [
        RotationalMotor {
          name "motor"
          maxVelocity 3.9667
          minPosition -1.5708
          maxPosition 1.5708
          maxTorque 1.1
        }
      ]
      endPoint Solid {
        translation -9.60178e-08 9.52021e-11 -0.03
        rotation 0.000317567 1 3.1345e-05 -8.19638e-08
        children [
          DEF GREY_BOX_TRANS Transform {
            translation 0 0 -0.0135
            children [
              Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 0.752941 0.752941 0.752941
                  }
                }
                geometry Box {
                  size 0.015 0.0612 0.047
                }
              }
            ]
          }
          DEF FRONT_CONNECTOR Connector {
            translation 0 0 -0.037
            rotation 0 1 0 3.1416
            children [
              USE ROUND_SHAPE
            ]
            name "front_connector"
            isLocked TRUE
            distanceTolerance 0.05
            axisTolerance 0.5
            rotationTolerance 0.5
          }
        ]
        boundingObject USE GREY_BOX_TRANS
        physics Physics {
          density -1
          mass 0.016
        }
      }
    }
  ]
  name "module_6"
  boundingObject USE BODY_GROUP
  physics USE BODY_PHYSICS
  controller "LocomotionController"
}
DEF MODULE_7 Robot {
  translation 9.81156e-08 0.0269877 0.582
  rotation 7.94294e-06 -7.97923e-06 1 1.5708
  children [
    USE BODY_GROUP
    DEF REAR_CONNECTOR Connector {
      translation 0 0 0.03
      children [
        DEF ROUND_SHAPE Transform {
          rotation 1 0 0 1.5708
          children [
            Shape {
              appearance Appearance {
                material Material {
                  diffuseColor 1 1 1
                }
              }
              geometry Cylinder {
                height 0.0005
                radius 0.0075
              }
            }
          ]
        }
      ]
      name "rear_connector"
      isLocked TRUE
      distanceTolerance 0.05
      axisTolerance 0.5
      rotationTolerance 0.5
    }
    DEF HINGE_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        position -5.11845e-08
        axis 0 1 0
        anchor 0 0 -0.03
      }
      device [
        RotationalMotor {
          name "motor"
          maxVelocity 3.9667
          minPosition -1.5708
          maxPosition 1.5708
          maxTorque 1.1
        }
      ]
      endPoint Solid {
        translation -1.59e-07 1.41363e-10 -0.03
        rotation 0.000257899 1 7.45451e-05 -5.11845e-08
        children [
          DEF GREY_BOX_TRANS Transform {
            translation 0 0 -0.0135
            children [
              Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 0.752941 0.752941 0.752941
                  }
                }
                geometry Box {
                  size 0.015 0.0612 0.047
                }
              }
            ]
          }
          DEF FRONT_CONNECTOR Connector {
            translation 0 0 -0.037
            rotation 0 1 0 3.1416
            children [
              USE ROUND_SHAPE
            ]
            name "front_connector"
            isLocked TRUE
            distanceTolerance 0.05
            axisTolerance 0.5
            rotationTolerance 0.5
          }
        ]
        boundingObject USE GREY_BOX_TRANS
        physics Physics {
          density -1
          mass 0.016
        }
      }
    }
  ]
  name "module_7"
  boundingObject USE BODY_GROUP
  physics USE BODY_PHYSICS
  controller "LocomotionController"
}
DEF MODULE_8 Robot {
  translation 9.38086e-08 0.026986 0.679
  rotation 1.15967e-05 -1.16479e-05 1 1.5708
  children [
    USE BODY_GROUP
    DEF REAR_CONNECTOR Connector {
      translation 0 0 0.03
      children [
        DEF ROUND_SHAPE Transform {
          rotation 1 0 0 1.5708
          children [
            Shape {
              appearance Appearance {
                material Material {
                  diffuseColor 1 1 1
                }
              }
              geometry Cylinder {
                height 0.0005
                radius 0.0075
              }
            }
          ]
        }
      ]
      name "rear_connector"
      isLocked TRUE
      distanceTolerance 0.05
      axisTolerance 0.5
      rotationTolerance 0.5
    }
    DEF HINGE_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        position -1.55942e-08
        axis 0 1 0
        anchor 0 0 -0.03
      }
      device [
        RotationalMotor {
          name "motor"
          maxVelocity 3.9667
          minPosition -1.5708
          maxPosition 1.5708
          maxTorque 1.1
        }
      ]
      endPoint Solid {
        translation -1.61937e-07 1.05852e-10 -0.03
        rotation 0.000139224 1 0.000183123 -1.55942e-08
        children [
          DEF GREY_BOX_TRANS Transform {
            translation 0 0 -0.0135
            children [
              Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 0.752941 0.752941 0.752941
                  }
                }
                geometry Box {
                  size 0.015 0.0612 0.047
                }
              }
            ]
          }
          DEF FRONT_CONNECTOR Connector {
            translation 0 0 -0.037
            rotation 0 1 0 3.1416
            children [
              USE ROUND_SHAPE
            ]
            name "front_connector"
            isLocked TRUE
            distanceTolerance 0.05
            axisTolerance 0.5
            rotationTolerance 0.5
          }
        ]
        boundingObject USE GREY_BOX_TRANS
        physics Physics {
          density -1
          mass 0.016
        }
      }
    }
  ]
  name "module_8"
  boundingObject USE BODY_GROUP
  physics USE BODY_PHYSICS
  controller "LocomotionController"
}
